catalog/nsili/catalog-nsili-sourcestoquery-ui/src/main/java/org/codice/alliance/catalog/nsili/sourcestoquery/ui/service/SourcesToQuery.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.nsili.sourcestoquery.ui.service;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import org.codice.alliance.nsili.endpoint.QuerySources;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+
+public class SourcesToQuery implements SourcesToQueryMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourcesToQuery.class);
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private BundleContext context;
+
+    private CatalogFramework catalogFramework;
+
+    private QuerySources querySources;
+
+    public void init() {
+        registerMbean();
+    }
+
+    @Override
+    public List<Map<String, String>> queriedSources() {
+        return toSourceIdMappedList(querySources.getQuerySources());
+    }
+
+    @Override
+    public List<Map<String, String>> queryableSources() {
+        return toSourceIdMappedList(catalogFramework.getSourceIds());
+    }
+
+    private List<Map<String, String>> toSourceIdMappedList(Collection<String> sourceIdList) {
```

Comment:
Assuming the hash map being created isn't being modified by the caller, you can simplify this method to:
```
private List<Map<String, String>> toSourceIdMappedList(Collection<String> sourceIdList) {

        return sourceIdList.stream()
                .map(s -> Collections.singletonMap("sourceId", s))
                .collect(Collectors.toList());
}
```